pkg/webui/console/views/device-general-settings/network-server-form/index.js

```diff
@@ -59,13 +68,48 @@ const NetworkServerForm = React.memo(props => {
   const formRef = React.useRef(null)
 
   const [error, setError] = React.useState('')
-  const [resetsFCnt, setResetsFCnt] = React.useState(
-    (isABP && device.mac_settings && device.mac_settings.resets_f_cnt) || false,
-  )
+
   const [lorawanVersion, setLorawanVersion] = React.useState(
     parseLorawanMacVersion(device.lorawan_version),
   )
 
+  const [deviceClasses, setDeviceClasses] = React.useState(() => {
+    const deviceClasses = [DEVICE_CLASSES.CLASS_A]
+
+    if (supports_class_b) {
+      deviceClasses.push(DEVICE_CLASSES.CLASS_B)
+    }
+
+    if (supports_class_c) {
+      deviceClasses.push(DEVICE_CLASSES.CLASS_C)
+    }
+
+    return deviceClasses
+  })
+  const handleDeviceClassChange = React.useCallback(evt => {
+    const { checked, name } = evt.target
+
+    let cls
+    if (name === 'supports_class_b') {
+      cls = DEVICE_CLASSES.CLASS_B
+    } else if (name === 'supports_class_c') {
+      cls = DEVICE_CLASSES.CLASS_C
+    }
+
+    if (checked && typeof cls !== 'undefined') {
+      setDeviceClasses(classes => [...classes, cls])
+    } else if (!checked && typeof cls !== 'undefined') {
+      setDeviceClasses(classes => classes.filter(c => c !== cls))
+    }
+  }, [])
+
+  let activationMode = ACTIVATION_MODES.ABP
+  if (supports_join) {
+    activationMode = ACTIVATION_MODES.OTAA
+  } else if (multicast) {
+    activationMode = ACTIVATION_MODES.MULTICAST
+  }
+
   const validationContext = React.useMemo(
```

Comment:
Same here. I don't think the composition of this context object should be of any performance concern.